modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -213,13 +214,7 @@ public ObjectRelationship createManyToManyObjectRelationshipTable(
 				objectRelationship);
 		}
 
-		User user = _userLocalService.getUser(userId);
-
-		objectRelationship.setDBTableName(
-			StringBundler.concat(
-				"R_", user.getCompanyId(), objectDefinition1.getShortName(),
-				"_", objectDefinition2.getShortName(), "_",
-				objectRelationship.getName()));
+		objectRelationship.setDBTableName(_generateDBTableName());
```

Comment:
@thallescezar as this method `_generateDBTableName` is only used once, we could inline to:

```
String dbTableName = null;

while (true) {
	StringBuilder sb = new StringBuilder(5);

	sb.append("R_");
	sb.append(StringUtil.toUpperCase(StringUtil.randomId(1)));
	sb.append(RandomUtil.nextInt(10));
	sb.append(StringUtil.toUpperCase(StringUtil.randomId(1)));
	sb.append(RandomUtil.nextInt(10));

	ObjectRelationship existingObjectRelationship =
		objectRelationshipPersistence.fetchByDTN_R(sb.toString(), false);

	if (existingObjectRelationship == null) {
		dbTableName = sb.toString();

		break;
	}
}

objectRelationship.setDBTableName(dbTableName);
```